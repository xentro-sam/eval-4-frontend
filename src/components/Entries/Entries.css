#entries {
    padding: 1vw;
    display: flex;
    justify-content: space-around;
    background-color: white;
    border-radius: 1vw;
    margin: 2vw 1vw;
}

#entry-actions {
    display: flex;
    flex-direction: row;
}

#edit-entry, #delete-entry {
    margin-left: 2vw;
}

#edit-entry > img, #delete-entry > img {
    width: 2vw;
    height: 2vw;
    cursor: pointer;
}

#entry {
    padding-left: 2.5vw;
}