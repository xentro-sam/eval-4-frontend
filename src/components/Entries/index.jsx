import * as React from 'react';
import './Entries.css';
import PropTypes from 'prop-types';
const editIcon = require('../../assets/user-pencil-write-ui-education@3x.png');
const deleteIcon = require('../../assets/trash-delete-recycle-bin-bucket-waste@3x.png');
import makeRequest from '../../utils/makeRequest';
import {DELETE_CONTENT_TYPE_ENTRY as deleteContentTypeEntry} from '../../constants/apiEndPoints';
import {v4 as uuidv4} from 'uuid';
import EntrySideModal from '../EntrySideModal';

export default function Entries(props) {
  const [showModal, setShowModal] = React.useState(false);
  const handleDelete = async () => {
    await makeRequest(deleteContentTypeEntry(props.contentTypeId, props.id), {});
    props.removeEntry(props.id);
  };
  return (
    <div id="entries">
      {props.reqAttributes.map((attribute) => {
        return (
          <div id="entry" key={uuidv4()}>
            {props[attribute]}
          </div>
        );
      })}
      <div id="entry-actions">
        <div id="edit-entry">
          <img src={editIcon} alt="edit" onClick={() => setShowModal(true)} />
        </div>
        <div id="delete-entry">
          <img src={deleteIcon} alt="delete" onClick={handleDelete} />
        </div>
      </div>
      <EntrySideModal show={showModal} onClose={() => setShowModal(false)} contentType={props.contentTypeName} fields={props.fields} updateEntry={props.updateEntry} api={props.api} task="Update" contentTypeId={props.contentTypeId} postId={props.id} />
    </div>
  );
}

Entries.propTypes = {
  id: PropTypes.number.isRequired,
  reqAttributes: PropTypes.array.isRequired,
  contentTypeId: PropTypes.number.isRequired,
  removeEntry: PropTypes.func.isRequired,
  contentTypeName: PropTypes.string.isRequired,
  fields: PropTypes.array.isRequired,
  updateEntry: PropTypes.func.isRequired,
  api: PropTypes.func.isRequired,
};
