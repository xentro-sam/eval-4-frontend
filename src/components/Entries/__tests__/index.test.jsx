import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Entries from '../index';
import {waitFor} from '@testing-library/dom';
import makeRequest from '../../../utils/makeRequest';
import {act} from 'react-dom/test-utils';

jest.mock('../../../utils/makeRequest');

const mockData = {
  'id': 1,
  'reqAttributes': ['name', 'description'],
  'contentTypeId': 1,
  'removeEntry': jest.fn(),
  'contentTypeName': 'mock title 1',
  'fields': ['name', 'description'],
  'updateEntry': jest.fn(),
  'api': jest.fn(),
  'name': 'name',
  'description': 'description',
};

describe('ContentType', () => {
  it('should render correctly', async () => {
    expect(makeRequest).not.toHaveBeenCalled();
    const {asFragment} = await act( async () => render(<Entries {...mockData} />));
    await waitFor(() => {
      expect(screen.getByText('name')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
