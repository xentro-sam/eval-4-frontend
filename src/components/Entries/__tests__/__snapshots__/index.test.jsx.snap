// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentType should render correctly 1`] = `
<DocumentFragment>
  <div
    id="entries"
  >
    <div
      id="entry"
    >
      name
    </div>
    <div
      id="entry"
    >
      description
    </div>
    <div
      id="entry-actions"
    >
      <div
        id="edit-entry"
      >
        <img
          alt="edit"
          src="user-pencil-write-ui-education@3x.png"
        />
      </div>
      <div
        id="delete-entry"
      >
        <img
          alt="delete"
          src="trash-delete-recycle-bin-bucket-waste@3x.png"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
